### ISSUE TITLE
Tensor aliasing support

### ISSUE THREAD

**Issue by karthikkattaopenxpu** [tenstorrent/tt-metal#35459]
### Is your feature request related to a problem? Please describe

vLLMs rely heavily on KV cache reuse across decoding steps.

- Keys and values are stored in large, preallocated buffers
- Each token appends into a slice of the same underlying tensor
- These slices are views (aliases), not new allocations

In PyTorch, indexing or slicing into a tensor does not create a new tensor by default.
Instead, it produces a view (alias) that references the same underlying storage.

```
t1 = torch.zeros((16), device="cuda")

with torch.inference_mode():
    t2 = torch.ones((16), device="cuda")
    t1[1] = t2[0]
```
Here:

t1[1] is not a new tensor. It is a view into t1’s storage. The assignment updates only a portion of t1, without reallocating it.

In the TT backend today, an operation like:
`
t1[1] = t2[0]`

may result in:

- Creation of a new tensor. 
- Loss of the original storage relationship
- Semantics diverging from PyTorch

This breaks correctness because: 

- Partial tensor updates rely on shared storage.
- Creating a new tensor invalidates in-place semantics.
- Subsequent reads of t1 may observe stale or incorrect data

### Describe the solution you'd like

To match PyTorch semantics, the backend shall support:

- Aliased views for indexing and slicing
- In-place partial updates without tensor reallocation
- Correct handling of view, reshape, and indexing ops
- Strict alias preservation in inference_mode

**Comment by aczajkowskiTT** [comment]
@karthikkattaopenxpu Would using the new upcoming narrow operation help you?
https://github.com/tenstorrent/tt-metal/pull/34905

**Comment by titash-openxpu** [comment]
This will help with partial extraction. Is there an API that supports partial updates of a tensor?

**Comment by aczajkowskiTT** [comment]
@ayerofieiev-tt Is there an API that supports partial updates of a tensor?

**Comment by aczajkowskiTT** [comment]
OXMIQ wants in-place modifications, not just view. Would be a big perf improvement

**Comment by fahmedchanTT** [comment]
@ayerofieiev-tt Any comments ?
